duckduckgo/iOS
# Use GRDB 2.1.0 packaged as xcframework

Task/Issue URL: https://app.asana.com/0/414235014887631/1204022201522438/f
Tech Design URL: https://app.asana.com/0/481882893211075/1204282785435610/f

**Description**:
Use GRDB as binary framework to avoid recompiling sqlite3.c every time the app is being built.
Because GRDB is a dependency of BSK which is a dependency of Core.framework (and not DuckDuckGo.app directly),
GRDB.framework is not available to be selected in “Copy Frameworks” phase nor is automatically copied
to the resulting app bundle. This change adds a custom Run Script build phase that handles copying and code signing
for GRDB.framework.

**Steps to test this PR**:
1. Smoke-test everything that uses plain sqlite database via GRDB (note that this has already been done once in https://github.com/duckduckgo/iOS/pull/1511).
1. Run the app on a physical device and verify that it doesn’t crash at startup due to a missing framework
1. Make adhoc build and install it on a device; verify that the app launches fine.

**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
